tests/routes/test_vespasearch.py

```diff
@@ -0,0 +1,743 @@
+        )
+        region = test_db.query(Geography).filter(Geography.id == parent_id).first()
+
+        params = {**base_params, **{"keyword_filters": {"regions": [region.slug]}}}
+        body_with_filters = _make_search_request(client, params=params)
+        filtered_family_slugs = [
+            f["family_slug"] for f in body_with_filters["families"]
+        ]
+        assert family["family_slug"] in filtered_family_slugs
+
+
+@pytest.mark.search
+@pytest.mark.parametrize("label,query", [("search", "the"), ("browse", "")])
+def test_invalid_keyword_filters(
+    label, query, test_vespa, test_db, monkeypatch, client
+):
+    monkeypatch.setattr(search, "_VESPA_CONNECTION", test_vespa)
+    _populate_search_db_families(test_db)
+
+    response = client.post(
+        SEARCH_ENDPOINT,
+        json={
+            "query_string": query,
+            "keyword_filters": {
+                "geographies": ["kenya"],
+                "unknown_filter_no1": ["BOOM"],
+            },
+        },
+    )
+    assert response.status_code == 422
+
+
+@pytest.mark.search
+@pytest.mark.parametrize(
+    "year_range", [(None, None), (1900, None), (None, 2020), (1900, 2020)]
+)
+def test_year_range_filterered_in(year_range, test_vespa, test_db, monkeypatch, client):
+    monkeypatch.setattr(search, "_VESPA_CONNECTION", test_vespa)
+    _populate_search_db_families(test_db)
+
+    # Search
+    params = {"query_string": "and", "year_range": year_range}
+    body = _make_search_request(client, params=params)
+    assert len(body["families"]) > 0
+
+    # Browse
+    params = {"query_string": "", "year_range": year_range}
+    body = _make_search_request(client, params=params)
+    assert len(body["families"]) > 0
+
+
+@pytest.mark.search
+@pytest.mark.parametrize("year_range", [(None, 2010), (2024, None)])
+def test_year_range_filterered_out(
+    year_range, test_vespa, test_db, monkeypatch, client
+):
+    monkeypatch.setattr(search, "_VESPA_CONNECTION", test_vespa)
+    _populate_search_db_families(test_db)
+
+    # Search
+    params = {"query_string": "and", "year_range": year_range}
+    body = _make_search_request(client, params=params)
+    assert len(body["families"]) == 0
+
+    # Browse
+    params = {"query_string": "", "year_range": year_range}
+    body = _make_search_request(client, params=params)
+    assert len(body["families"]) == 0
+
+
+@pytest.mark.search
+@pytest.mark.parametrize("label, query", [("search", "the"), ("browse", "")])
+def test_multiple_filters(label, query, test_vespa, test_db, monkeypatch, client):
+    monkeypatch.setattr(search, "_VESPA_CONNECTION", test_vespa)
+    _populate_search_db_families(test_db)
+
+    params = {
+        "query_string": query,
+        "keyword_filters": {
+            "countries": ["south-korea"],
+            "sources": ["CCLW"],
+            "categories": ["Legislative"],
+        },
+        "year_range": (1900, 2020),
+    }
+
+    _ = _make_search_request(client, params)
```

Comment:
No assertion?

Reply:
This one currently just tests that the request is successful (within `_make_search_request`), definetly opportunity to improve this test, but I'll leave it for today